src/ArrayValueCalculator.java

```diff
@@ -0,0 +1,28 @@
+public class ArrayValueCalculator {
+
+    public static int doCalc(String[][] matrix) {
+            if (matrix == null) {
+                throw new NullPointerException("The array must not be null, the array must be 4x4");
+            }
+            if (matrix.length != 4) {
```

Comment:
Would be better to extract this piece of code to a separate method - `validateMatrixSize(matrix)`
`if (matrix.length != 4) {
                throw new ArraySizeException("The array must be 4x4");
            }
            for (int i = 0; i < matrix.length; i++) {
                if (matrix[i].length != 4) {
                    throw new ArraySizeException("The array must be 4x4");
                }
            }`